salt/journal-cms/init.sls

```diff
@@ -297,22 +297,41 @@ site-was-installed-check:
             - cmd: site-install
 
 site-install:
-    cmd.run:
-        - name: |
-            set -e
-            ../vendor/bin/drush site-install minimal --existing-config -y
-            ####test -e /home/{{ pillar.elife.deploy_user.username }}/site-was-installed.flag && ../vendor/bin/drush cr || echo "site was not installed before, not rebuilding cache"
-            #../vendor/bin/drush cr # may fail with "You have requested a non-existent service "cache.backend.redis"
-            redis-cli flushall
-        - cwd: /srv/journal-cms/web
-        - runas: {{ pillar.elife.deploy_user.username }}
-        - require:
-            - journal-cms-repository
-        # always perform a new site-install on dev and ci
-        {% if pillar.elife.env not in ['dev', 'ci'] %}
-        - unless:
-            - sudo -u {{ pillar.elife.deploy_user.username}} ../vendor/bin/drush cget system.site name
-        {% endif %}
+  cmd.run:
+    - name: |
+        set -e
+        ../vendor/bin/drush site-install minimal --existing-config -y
+        ####test -e /home/{{ pillar.elife.deploy_user.username }}/site-was-installed.flag && ../vendor/bin/drush cr || echo "site was not installed before, not rebuilding cache"
+        #../vendor/bin/drush cr # may fail with "You have requested a non-existent service "cache.backend.redis"
+        redis-cli flushall
+    - cwd: /srv/journal-cms/web
+    - runas: {{ pillar.elife.deploy_user.username }}
+    - require:
+        - journal-cms-repository
+
+# Always perform a new site-install on dev and ci, never on continuumtest or prod
+{% if pillar.elife.env in ['continuumtest', 'prod'] %}
+site-install_skip:  # Define a new state to skip site-install on continuumtest and prod
+  cmd.run:
+    - name: echo "Never perform site-install on continuumtest and prod"
+{% elif pillar.elife.env in ['dev', 'ci'] %}
+site-install_dev_ci:  # Define a new state to perform site-install on dev and ci
+  cmd.run:
+    - name: |
+        set -e
+        ../vendor/bin/drush site-install minimal --existing-config -y
+        ####test -e /home/{{ pillar.elife.deploy_user.username }}/site-was-installed.flag && ../vendor/bin/drush cr || echo "site was not installed before, not rebuilding cache"
+        #../vendor/bin/drush cr # may fail with "You have requested a non-existent service "cache.backend.redis"
+        redis-cli flushall
+    - cwd: /srv/journal-cms/web
+    - runas: {{ pillar.elife.deploy_user.username }}
+    - require:
+        - journal-cms-repository
+{% else %}
+site-install_other:  # Define a new state to check site-install condition
```

Comment:
This other condition needs to retain the ability to trigger a reinstall if `sudo -u {{ pillar.elife.deploy_user.username }} ../vendor/bin/drush cget system.site name` indicates that the site has not been setup.